### ISSUE TITLE
solaar: _setCidReporting(): hidpp20.py:575:_setCidReporting:TypeError: 'NoneType' object is not subscriptable 

### ISSUE THREAD

**Issue by rathann** [pwr-Solaar/Solaar#1238]
**Information**

- Solaar version (`solaar --version` or `git describe --tags` if cloned from this repository): **1.0.6**
- Distribution: **Fedora 34**
- Kernel version (ex. `uname -srmo`): **5.12.13-300.fc34.x86_64**
- Output of `solaar show`:

<details>

```
Wired Devices
  1: MX Master 3 Wireless Mouse
     Device path  : /dev/hidraw28
     USB id       : 046d:B023
     Codename     : MX Master 3
     Kind         : mouse
     Protocol     : HID++ 4.5
     Serial number: ?
     Model ID:      B02340820000
     Unit ID:       020BFDBC
        Bootloader: BOT 95.01.B0015
          Firmware: MPM 19.01.B0015
             Other: 
     Supports 31 HID++ 2.0 features:
         0: ROOT                   {0000}   
         1: FEATURE SET            {0001}   
         2: DEVICE FW VERSION      {0003}   
            Firmware: Bootloader BOT 95.01.B0015 00006A9CA89D
            Firmware: Firmware MPM 19.01.B0015 B0236A9CA89D
            Firmware: Other   
            Unit ID: 020BFDBC  Model ID: B02340820000  Transport IDs: {'btleid': 'B023', 'wpid': '4082'}
         3: DEVICE NAME            {0005}   
            Name: Wireless Mouse MX Master 3
            Kind: mouse
         4: WIRELESS DEVICE STATUS {1D4B}   
         5: RESET                  {0020}   
         6: CRYPTO ID              {0021}   
         7: DEVICE FRIENDLY NAME   {0007}   
         8: BATTERY STATUS         {1000}   
            Battery: 100%, discharging, next level 50%.
         9: REPROG CONTROLS V4     {1B04}   
            Key/Button Actions (saved): {'195': 195, '196': 196, '80': 80, '81': 81, '82': 82, '83': 83, '86': 86}
            Key/Button Actions        : {'80': 80, '81': 81, '82': 82, '83': 83, '86': 86, '195': 195, '196': 196}
        10: CHANGE HOST            {1814}   
            Change Host        : 1:fedora
        11: XY STATS               {2250}   
        12: ADJUSTABLE DPI         {2201}   
            Sensitivity (DPI) (saved): 1000
            Sensitivity (DPI)        : 1000
        13: SMART SHIFT            {2110}   
            Scroll Wheel Rachet (saved): 10
            Scroll Wheel Rachet        : 10
        14: HIRES WHEEL            {2121}   
            Multiplier: 15
            Has invert: Normal wheel motion
            Has ratchet switch: Normal wheel mode
            High resolution mode
            HID notification
            Scroll Wheel Direction (saved): False
            Scroll Wheel Direction        : False
            Scroll Wheel Resolution (saved): True
            Scroll Wheel Resolution        : True
        15: THUMB WHEEL            {2150}   
            Thumb Wheel HID++ Scrolling (saved): False
            Thumb Wheel HID++ Scrolling        : False
            Thumb Wheel Direction (saved): False
            Thumb Wheel Direction        : False
        16: WHEEL STATS            {2251}   
        17: DFUCONTROL SIGNED      {00C2}   
        18: DEVICE RESET           {1802}   internal, hidden
        19: unknown:1803           {1803}   internal, hidden
        20: CONFIG DEVICE PROPS    {1806}   internal, hidden
        21: unknown:1813           {1813}   internal, hidden
        22: OOBSTATE               {1805}   internal, hidden
        23: unknown:1830           {1830}   internal, hidden
        24: unknown:18A1           {18A1}   internal, hidden
        25: unknown:1E00           {1E00}   hidden
        26: unknown:1EB0           {1EB0}   internal, hidden
        27: unknown:1861           {1861}   internal, hidden
        28: unknown:9300           {9300}   internal, hidden
        29: unknown:9001           {9001}   internal, hidden
        30: unknown:9205           {9205}   internal, hidden
     Has 8 reprogrammable keys:
         0: Left Button               , default: Left Click                  => Left Click                
             mse, analytics key events, pos:0, group:1, group mask:g1
             reporting: default
         1: Right Button              , default: Right Click                 => Right Click               
             mse, analytics key events, pos:0, group:1, group mask:g1
             reporting: default
         2: Middle Button             , default: Mouse Middle Button         => Mouse Middle Button       
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         3: Back Button               , default: Mouse Back Button           => Mouse Back Button         
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:2, group mask:g1,g2
             reporting: default
         4: Forward Button            , default: Mouse Forward Button        => Mouse Forward Button      
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:2, group mask:g1,g2
             reporting: default
         5: Mouse Gesture Button      , default: Gesture Button Navigation   => Gesture Button Navigation 
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         6: Smart Shift               , default: Smart Shift                 => Smart Shift               
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         7: Virtual Gesture Button    , default: Virtual Gesture Button      => Virtual Gesture Button    
             divertable, virtual, raw XY, force raw XY, pos:0, group:4, group mask:empty
             reporting: default
     Battery: 100%, discharging, next level 50%.
```
</details>

**Describe the bug**
solaar throws an exception:
```
Truncated backtrace:
hidpp20.py:575:_setCidReporting:TypeError: 'NoneType' object is not subscriptable

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/solaar/listener.py", line 384, in _process_receiver_event
    _start(device_info)
  File "/usr/lib/python3.9/site-packages/solaar/listener.py", line 287, in _start
    rl = ReceiverListener(receiver, _status_callback)
  File "/usr/lib/python3.9/site-packages/solaar/listener.py", line 81, in __init__
    receiver.status.changed(True)
  File "/usr/lib/python3.9/site-packages/logitech_receiver/status.py", line 310, in changed
    _settings.apply_all_settings(d)
  File "/usr/lib/python3.9/site-packages/logitech_receiver/settings.py", line 1167, in apply_all_settings
    s.apply()
  File "/usr/lib/python3.9/site-packages/logitech_receiver/settings.py", line 186, in apply
    self.write(value, save=False)
  File "/usr/lib/python3.9/site-packages/logitech_receiver/settings.py", line 263, in write
    reply = self._rw.write(self._device, int(key), data_bytes)
  File "/usr/lib/python3.9/site-packages/logitech_receiver/settings_templates.py", line 659, in write
    key_struct.set_diverted(data_bytes == b'\x01')
  File "/usr/lib/python3.9/site-packages/logitech_receiver/hidpp20.py", line 473, in set_diverted
    self._setCidReporting(flags=flags)
  File "/usr/lib/python3.9/site-packages/logitech_receiver/hidpp20.py", line 575, in _setCidReporting
    if _unpack('!BBBBB', ret[:5]) != pkt and _log.isEnabledFor(_WARNING):
TypeError: 'NoneType' object is not subscriptable

Local variables in innermost frame:
self: <logitech_receiver.hidpp20.ReprogrammableKeyV4 object at 0x7f52845f7280>
flags: {NamedInt(1, 'diverted'): False}
remap: 0
FLAG_TO_CAPABILITY: {NamedInt(1, 'diverted'): NamedInt(32, 'divertable'), NamedInt(4, 'persistently diverted'): NamedInt(64, 'persistently divertable'), NamedInt(256, 'analytics key events reporting'): NamedInt(1024, 'analytics key events'), NamedInt(64, 'force raw XY diverted'): NamedInt(512, 'force raw XY'), NamedInt(16, 'raw XY diverted'): NamedInt(256, 'raw XY')}
bfield: 2
f: NamedInt(1, 'diverted')
v: False
pkt: (0, 86, 2, 0, 0)
ret: None
```

**To Reproduce**

Unknown, will provide once downstream reporter responds.

**Comment by pfps** [comment]
This has almost certainly been fixed in commit ed0b730e.  What has likely happened is that a request to the device has timed out but the code that threw an exception didn't take this situation into account.

**Comment by pfps** [comment]
Closing as likely fixed and no response from original reported.
